src/globus_cli/parsing/param_types/guest_activity_notify_param.py

```diff
@@ -0,0 +1,171 @@
+import typing as t
+
+import click
+from click.shell_completion import CompletionItem
+
+from globus_cli.constants import ExplicitNullType
+
+
+class GCSManagerGuestActivityNotificationParamType(click.ParamType):
+
+    VALID_STATUSES = {
+        "succeeded",
+        "failed",
+    }
+
+    VALID_TRANSFER_USES = {
+        "source",
+        "destination",
+    }
+
+    VALID_NOTIFICATION_VALUES = VALID_TRANSFER_USES.copy()
+    VALID_NOTIFICATION_VALUES |= VALID_STATUSES
+
+    def get_metavar(self, param: click.Parameter) -> str:
+        return "{all,succeeded,failed,source,destination}"
+
+    def convert(
+        self,
+        value: str,
+        param: t.Union[click.Parameter, None],
+        ctx: t.Union[click.Context, None],
+    ) -> t.Union[t.Dict[str, t.List[str]], ExplicitNullType]:
+
+        if value == "":
+            return {
+                "status": [],
+                "transfer_use": [],
+            }
+
+        if value.lower() == "all":
+            return {
+                "status": sorted(self.VALID_STATUSES),
+                "transfer_use": sorted(self.VALID_TRANSFER_USES),
+            }
+
+        policy: t.Dict[str, t.List[str]] = {
+            "status": [],
+            "transfer_use": [],
+        }
+
+        # ignore white space, parse input sans case-sensitivity
+        lowercase_vals: t.Set[str] = {s.strip().lower() for s in value.split(",")}
+
+        if "all" in lowercase_vals:
+            raise click.UsageError(
+                '--activity-notifications cannot accept "all" with other values'
+            )
+
+        val: str
+        if lowercase_vals <= self.VALID_NOTIFICATION_VALUES:
+            for val in lowercase_vals:
+                if val in self.VALID_TRANSFER_USES:
+                    policy["transfer_use"].append(val)
+                else:
+                    policy["status"].append(val)
+        else:
+            invalid_values = sorted(lowercase_vals - self.VALID_NOTIFICATION_VALUES)
+            raise click.UsageError(
+                "--activity-notifications received these invalid values: "
+                f"{invalid_values}"
+            )
+
+        # Fill in implied values.
+        k: str
+        v: t.List[str]
+        for k, v in policy.items():
+            if k == "transfer_use" and not v:
+                v = list(self.VALID_TRANSFER_USES)
+            elif k == "status" and not v:
+                v = list(self.VALID_STATUSES)
+            # Ensure deterministic ordering for testability.
+            policy[k] = sorted(v)
```

Comment:
There are three minor changes I'd like here.

One is that I think the comment at the top about "implied values" should be a bit more explicit about what/why it's implied. I've suggested what I think is right for that but please review/tweak as appropriate.

The second is that the type of `policy` is declared above, so we shouldn't need to declare the types of `k` and `v`.

The last is that `policy[k] = sorted(v)` looks strange to me because we're basically doing `sorted(list(...))` and we can remove one of the two. I've suggested _a_ way of altering it, but others (e.g., `v = sorted(self.VALID_...)`) are also possible.

```suggestion
        # Fill in implied values.
        # If statuses were given but no uses, all uses are listed.
        # If uses were given but no statuses, all statuses are listed.
        for k, v in policy.items():
            # use sorted lists for deterministic ordering, for better testability
            if k == "transfer_use" and not v:
                policy[k] = sorted(self.VALID_TRANSFER_USES)
            elif k == "status" and not v:
                policy[k] = sorted(self.VALID_STATUSES)
```